### DISCUSSION TITLE
Can't load file processed on other machine

### DISCUSSION THREAD

**Discussion by atale480** [pyg-team/pytorch_geometric#5987] [Q&A] [answered]
I'm working on 3d Meshes. I have 380 meshes of dataset ' A Benchmark for 3D Mesh Segmentation '. Im doing some preprocessing on those mesh. It was taking too long hence I did some file preprocessing on my local system and remaining on google colab. Later I downloaded all prepocessed files from my drive and merged all preprocessed files ( i.e. from my local system and downloaded ). When I tried to load downloaded files from colab using torch.load then it is throwing error.

>`---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
/Users/rohanatale/Thesis/Notebooks/GCN_on_Mesh.ipynb Cell 10 in <cell line: 1>()
----> 1 a = torch.load('/Users/rohanatale/M2G/data_35.pt')

>File ~/miniforge3/envs/pytorch/lib/python3.8/site-packages/torch/serialization.py:723, in load(f, map_location, pickle_module, **pickle_load_args)
    721             opened_file.seek(orig_position)
    722             return torch.jit.load(opened_file, map_location=map_location)
--> 723         return _load(opened_zipfile, map_location, pickle_module, **pickle_load_args)
    724 return _legacy_load(opened_file, map_location, pickle_module, **pickle_load_args)

>File ~/miniforge3/envs/pytorch/lib/python3.8/site-packages/torch/serialization.py:1060, in _load(zip_file, map_location, pickle_module, pickle_file, **pickle_load_args)
   1058 unpickler = UnpicklerWrapper(data_file, **pickle_load_args)
   1059 unpickler.persistent_load = persistent_load
-> 1060 result = unpickler.load()
   1062 torch._utils._validate_loaded_sparse_tensors()
   1064 return result

>File ~/miniforge3/envs/pytorch/lib/python3.8/site-packages/torch/serialization.py:1053, in _load.<locals>.UnpicklerWrapper.find_class(self, mod_name, name)
   1051         pass
   1052 mod_name = load_module_mapping.get(mod_name, mod_name)
-> 1053 return super().find_class(mod_name, name)

>AttributeError: Can't get attribute 'DataEdgeAttr' on <module 'torch_geometric.data.data' from '/Users/rohanatale/miniforge3/envs/pytorch/lib/python3.8/site-packages/torch_geometric/data/data.py'>`

But when I load file processed on my local system those work fine. 
Does all files need to be processed on same system?
Am I missing something?

I'm local system is MacBook Air M1.

**Comment by rusty1s** [comment, marked as answer]
Looks like you are using different PyG versions on both systems. Can you confirm?

**Reply by atale480** [reply]
Package versions on My MacBook M1

> torch                         =         1.13.0.dev20220707 
> torch-geometric        =       2.0.4                   
> torch-scatter                =    2.0.9
> torch-sparse                 =   0.6.14                   
> torchaudio                       = 0.14.0.dev20220603         
> torchvision                      = 0.14.0.dev20220707          

Package versions on Google Colab

> torch                         =         1.12.1+cu113
> torch-geometric        =       2.1.0                   
> torch-scatter                =    2.0.9
> torch-sparse                 =   0.6.15                
> torchaudio                       = 0.12.1+cu113     
> torchvision                      = 0.13.1+cu113


If My issues are causing due to different version then if I preprocessed all files on colab then would I be able to load that file on MacBook M1 system?

or do I need to make same version for all packages on MacBook M1 as that of colab and then my program will work

**Reply by rusty1s** [reply]
It just needs the same version of PyG.

**Reply by atale480** [reply]
Worked like charm thank you sir !!!
